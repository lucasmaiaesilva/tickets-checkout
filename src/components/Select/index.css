.select select {
   background: url(http://i62.tinypic.com/15xvbd5.png) no-repeat 96% 0;
   height: 29px;
   overflow: hidden;
   background: transparent;
   border: none;
   font-size: 14px;
   height: 29px;
   padding: 5px; /* If you add too much padding here, the options won't show in IE */
   width: 68px;
   -webkit-appearance: button;
   -webkit-border-radius: 2px;
   -webkit-box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.1);
   -webkit-padding-end: 20px;
   -webkit-padding-start: 2px;
   -webkit-user-select: none;
   background-image: url(http://i62.tinypic.com/15xvbd5.png), -webkit-linear-gradient(#FAFAFA, #F4F4F4 40%, #E5E5E5);
   background-position: 80% center;
   background-repeat: no-repeat;
   border: 1px solid #AAA;
   color: #555;
   font-size: inherit;
   margin: 20px;
   overflow: hidden;
   padding: 5px 10px;
   text-overflow: ellipsis;
   white-space: nowrap;
   width: 74px;
  outline: none;
}

