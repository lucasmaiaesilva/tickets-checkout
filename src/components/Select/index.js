import React from 'react';

import './index.css';

const Select = ({ quantities, ticketId, disabled, handleChange, price, discount }) => (
  <div className="select">
    <select disabled={disabled} onChange={(e) => handleChange(price, ticketId, e, discount)}>
      <option value={0}>-</option>
      {quantities.map(item => (
        <option key={item} value={item}>{item}</option>
      ))}
    </select>
  </div>
);

Select.defaultProps = {
  disabled: false 
};

export default Select;

