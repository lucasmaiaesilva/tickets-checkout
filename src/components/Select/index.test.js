import React from 'react';
import { shallow, mount } from 'enzyme';
import Select from './';

describe('[Component] Button', () => {
  const props = {
    quantities: [2, 3, 4, 5],
    ticketId: 'asd',
    handleChange: jest.fn(),
    price: 400,
    discount: 2,
  };
  it('render correctly', () => {
    const wrapper = shallow(<Select {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render correctly when disabled', () => {
    const wrapper = shallow(<Select {...props} disabled />);
    expect(wrapper).toMatchSnapshot();
  });
  it('test handleChange callback', () => {
    const wrapper = mount(<Select {...props} />);
    wrapper.find('select').simulate('change');
    expect(props.handleChange).toHaveBeenCalled();
  });
});

