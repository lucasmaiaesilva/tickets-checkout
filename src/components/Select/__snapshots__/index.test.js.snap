// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Button render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    disabled={false}
    discount={2}
    handleChange={[Function]}
    price={400}
    quantities={
        Array [
            2,
            3,
            4,
            5,
          ]
    }
    ticketId="asd"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        disabled={false}
        onChange={[Function]}
>
        <option
                value={0}
        >
                -
        </option>
        <option
                value={2}
        >
                2
        </option>
        <option
                value={3}
        >
                3
        </option>
        <option
                value={4}
        >
                4
        </option>
        <option
                value={5}
        >
                5
        </option>
</select>,
      "className": "select",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value={0}
>
            -
</option>,
          Array [
            <option
              value={2}
>
              2
</option>,
            <option
              value={3}
>
              3
</option>,
            <option
              value={4}
>
              4
</option>,
            <option
              value={5}
>
              5
</option>,
          ],
        ],
        "disabled": false,
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "value": 0,
          },
          "ref": null,
          "rendered": "-",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "value": 2,
          },
          "ref": null,
          "rendered": 2,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "value": 3,
          },
          "ref": null,
          "rendered": 3,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": 4,
            "value": 4,
          },
          "ref": null,
          "rendered": 4,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "value": 5,
          },
          "ref": null,
          "rendered": 5,
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          disabled={false}
          onChange={[Function]}
>
          <option
                    value={0}
          >
                    -
          </option>
          <option
                    value={2}
          >
                    2
          </option>
          <option
                    value={3}
          >
                    3
          </option>
          <option
                    value={4}
          >
                    4
          </option>
          <option
                    value={5}
          >
                    5
          </option>
</select>,
        "className": "select",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value={0}
>
              -
</option>,
            Array [
              <option
                value={2}
>
                2
</option>,
              <option
                value={3}
>
                3
</option>,
              <option
                value={4}
>
                4
</option>,
              <option
                value={5}
>
                5
</option>,
            ],
          ],
          "disabled": false,
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "-",
              "value": 0,
            },
            "ref": null,
            "rendered": "-",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "value": 2,
            },
            "ref": null,
            "rendered": 2,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "value": 3,
            },
            "ref": null,
            "rendered": 3,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "value": 4,
            },
            "ref": null,
            "rendered": 4,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "value": 5,
            },
            "ref": null,
            "rendered": 5,
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`[Component] Button render correctly when disabled 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    disabled={true}
    discount={2}
    handleChange={[Function]}
    price={400}
    quantities={
        Array [
            2,
            3,
            4,
            5,
          ]
    }
    ticketId="asd"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        disabled={true}
        onChange={[Function]}
>
        <option
                value={0}
        >
                -
        </option>
        <option
                value={2}
        >
                2
        </option>
        <option
                value={3}
        >
                3
        </option>
        <option
                value={4}
        >
                4
        </option>
        <option
                value={5}
        >
                5
        </option>
</select>,
      "className": "select",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value={0}
>
            -
</option>,
          Array [
            <option
              value={2}
>
              2
</option>,
            <option
              value={3}
>
              3
</option>,
            <option
              value={4}
>
              4
</option>,
            <option
              value={5}
>
              5
</option>,
          ],
        ],
        "disabled": true,
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "value": 0,
          },
          "ref": null,
          "rendered": "-",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "value": 2,
          },
          "ref": null,
          "rendered": 2,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "value": 3,
          },
          "ref": null,
          "rendered": 3,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": 4,
            "value": 4,
          },
          "ref": null,
          "rendered": 4,
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "value": 5,
          },
          "ref": null,
          "rendered": 5,
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          disabled={true}
          onChange={[Function]}
>
          <option
                    value={0}
          >
                    -
          </option>
          <option
                    value={2}
          >
                    2
          </option>
          <option
                    value={3}
          >
                    3
          </option>
          <option
                    value={4}
          >
                    4
          </option>
          <option
                    value={5}
          >
                    5
          </option>
</select>,
        "className": "select",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value={0}
>
              -
</option>,
            Array [
              <option
                value={2}
>
                2
</option>,
              <option
                value={3}
>
                3
</option>,
              <option
                value={4}
>
                4
</option>,
              <option
                value={5}
>
                5
</option>,
            ],
          ],
          "disabled": true,
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "-",
              "value": 0,
            },
            "ref": null,
            "rendered": "-",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "value": 2,
            },
            "ref": null,
            "rendered": 2,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "value": 3,
            },
            "ref": null,
            "rendered": 3,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "value": 4,
            },
            "ref": null,
            "rendered": 4,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "value": 5,
            },
            "ref": null,
            "rendered": 5,
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
